### DISCUSSION TITLE
Piping output to s3 seems to produce unreadable files

### DISCUSSION THREAD

**Discussion by warflash** [ImageMagick/ImageMagick#5434] [Help]
Hey 👋

I'm trying to write a powershell script that takes an input path, converts the image to .webp and pipes the output into the aws cli so it doesn't have to get saved to disk first.

The command I'm using looks like this: 
`magick path_to_image.png webp:- | aws s3 --endpoint=xx cp - s3_path.webp `

The issue I'm having is that magick seems to pipe invalid formats as the files that end up on s3 do have the expected size, they just are corrupted when opening them.

Appreciate any help on this!

**Comment by fmw42** [comment]
What do you get from magick -version?  Do you have webp installed?

**Reply by warflash** [reply]
My magick version is : `Version: ImageMagick 7.1.0-45 Q16-HDRI x64 e32676e:20220731 https://imagemagick.org`
I'm not 100% certain what you refer to with having webp installed - but converting to webp and saving to disk does work + changing the target format to .png or similar also results in a broken format.

Played around with it a bit and tried the example from the docs:
`magick path_to_image.jpg png:- | magick - -shade 110x90  test.png`
which results in: `magick.exe: no decode delegate for this image format `' @ error/constitute.c/ReadImage/741.`

What's very confusing is that the same exact command runs just fine in windows cmd but errors in powershell🤔

**Comment by fmw42** [comment]
What do you get from

magick -version

Does the list of Delegates include webp?

**Reply by warflash** [reply]
Yes it does include webp.
`Version: ImageMagick 7.1.0-45 Q16-HDRI x64 e32676e:20220731 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193231332)`

In the comment above I mentioned windows cmd works for that input but powershell fails. To be honest I have no clue what the reason for that would be and whether thats on microsoft or on magick and how it could be resolved

**Comment by fmw42** [comment]
Does a simple convert in powershell fail?  Do you get the same version information in Powershell when you run magick -version?

**Reply by warflash** [reply]
I do get the same version output in powershell yeah. Simple conversions do work fine in ps, just verified that

**Reply by fmw42** [reply]
So that looks like an issue of Powershell to aws

**Reply by warflash** [reply]
I dont think so as `magick path_to_image.jpg png:- | magick - -shade 110x90 test.png` also fails in powershell. I dont have a second windows installation to verify whether it's exclusive to me but piping magick through powershell in general does seem to have issues

**Reply by fmw42** [reply]
Sorry, I do not know how piping works in powershell

**Reply by warflash** [reply]
No worries, me neither.. Thanks a lot for the assistance tho, much appreciated🎉

**Comment by snibgo** [comment]
How are you running `magick` in your Powershell script? `magick` isn't a cmdlet, and it doesn't send Powershell objects down the pipe. See, for example, https://stackoverflow.com/questions/5317269/how-to-use-cmd-type-pipe-piping-in-powershell

**Reply by warflash** [reply]
Oh thats interesting, was not aware of that. I guess that explains the behaviour then^^ I was running it using `invoke-expression -command "xx"`. Guess I'll migrate to magick.net or something similar then. Was trying to save time going the ps way but that kind of backfired.. 
Thanks a lot for the info tho, learned something new today 👌
